dask/dataframe/io/parquet/core.py

```diff
@@ -381,12 +388,17 @@ def to_parquet(
     if division_info["name"] is None:
         # As of 0.24.2, pandas will rename an index with name=None
         # when df.reset_index() is called.  The default name is "index",
-        # (or "level_0" if "index" is already a column name)
-        division_info["name"] = "index" if "index" not in df.columns else "level_0"
+        # but dask will always change the name to the NONE_LABEL constant
+        if NONE_LABEL not in df.columns:
+            division_info["name"] = NONE_LABEL
+        else:
+            raise ValueError(
+                "Must specify an index if __null_dask_index__ is a column name"
+            )
```

Comment:
Are these lines covered by tests?

Reply:
Good call - just added coverage and improved the message a bit.